### PR TITLE
Collect Awarded supplier and contract details

### PR DESCRIPTION
**Pull request by katstevens**
Ticket: https://trello.com/c/kGVfzXuH/749-3-collect-award-status-on-buyer-dashboard

PR 4 of 4: Briefs front end.

See also:
PR 1 of 4: API: https://github.com/alphagov/digitalmarketplace-api/pull/622
PR 2 of 4: API client: https://github.com/alphagov/digitalmarketplace-apiclient/pull/82 (merged)
PR 3 of 4: Question content: https://github.com/alphagov/digitalmarketplace-frameworks/pull/464 (merged)

What's included:
- Two new routes for the 'yes' path: `award_brief` (collecting the winning supplier) and `award_brief_details` (collecting the contract details).
- New custom WTForm for displaying an alphabetical list of suppliers, based on the Brief's BriefResponses
- Templates for the new routes: `award.html` and `award_details.html`
- Display Briefs with status 'awarded' in the 'Closed Requirements' section of the buyer dashboard
- Include the new question manifest (see PR 3 of 4) for use with the content loader
- Replace the deprecated `macros/toolkit_forms.html` with `toolkit/forms/macros/forms.html`

Changes required for new Brief/BriefResponse statuses:
- Allow buyers to view/download BriefResponses for `awarded` Briefs as well as `closed`
- Consider what steps to display on the Brief overview page for a new status (awarded Briefs are treated as closed for now)

**Required before merging:** 
- ~~Content for first awards page when a Brief has had no BriefResponses~~
- ~~the API client and Frameworks versions needs to be bumped before this PR can be merged.~~
All ready! However this shouldn't be deployed before the API.

### CODE REVIEW

File: app/main/views/buyers.py [hunk 1]
```diff
@@ -345,6 +346,129 @@ def view_brief_responses(framework_slug, lot_slug, brief_id):
...
+                form=form,
+                form_errors=form_errors,
+            ), 400
+
+        if form.data:
+            try:
+                data_api_client.update_brief_award_brief_response(
+                    brief_id,
+                    form.data['supplier'],
+                    updated_by=current_user.email_address
```
Comment (benvand) [inline app/main/views/buyers.py]: This wasn't working for me, had to change it like:
```
 -    updated_by=current_user.email_address
 +    current_user.email_address
```

File: app/main/views/buyers.py [hunk 2]
```diff
@@ -345,6 +346,129 @@ def view_brief_responses(framework_slug, lot_slug, brief_id):
...
+                url_for(
+                    ".award_brief_details",
+                    framework_slug=brief['frameworkSlug'],
+                    lot_slug=brief['lotSlug'],
+                    brief_id=brief['id'],
+                    brief_response_id=form.data['supplier']
+                )
+            )
+
+    else:
```
Comment (benvand) [inline app/main/views/buyers.py]: Looks like this `else` is not strictly required?

File: tests/main/views/test_buyers.py [hunk 3]
```diff
@@ -3098,3 +3144,328 @@ def test_do_not_show_question_and_answer_dates_for_closed_brief(self, data_api_c
...
+            assert self.data_api_client.update_brief_award_details.call_args == mock.call(
+                '1234', '5678',
+                {'awardedContractStartDate': "2020-12-31", "awardedContractValue": "88.84"},
+                updated_by="buyer@email.com"
+            )
+            assert res.status_code == 302
+            assert res.location == "http://localhost/buyers"
+            self.assert_flashes("updated-brief")
+
+    def _setup_api_error_response(self, error_json):
```
Comment (benvand) [inline tests/main/views/test_buyers.py]: I find it easier to find helpers if they're at the top of the class

Comment (katstevens) [inline tests/main/views/test_buyers.py]: I don't have a strong opinion on this but I tend to keep helpers near the tests that use them. However in this case the class isn't that big so it doesn't really make a difference 😄 I'll move it underneath the setup/teardown.

File: app/main/__init__.py [hunk 4]
```diff
@@ -15,6 +15,7 @@
 content_loader.load_manifest('digital-outcomes-and-specialists-2', 'briefs', 'edit_brief')
 content_loader.load_manifest('digital-outcomes-and-specialists-2', 'brief-responses', 'output_brief_response')
 content_loader.load_manifest('digital-outcomes-and-specialists-2', 'clarification_question', 'clarification_question')
+content_loader.load_manifest('digital-outcomes-and-specialists-2', 'briefs', 'award_brief')
```
Comment (idavidmcdonald) [inline app/main/__init__.py]: Will need the same for DOS1

File: app/main/forms/awards.py [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+        """
+            Requires extra keyword arguments:
+             - `supplier_list` - list of suppliers for the multiple choice
+        """
+        super(AwardedSupplierForm, self).__init__(*args, **kwargs)
+
+        # popping this kwarg so we don't risk it getting fed to wtforms default implementation which might use it
+        # as a data field if there were a name collision
+        suppliers = kwargs.pop("suppliers", [])
+        self.supplier.choices = [(s['id'], s['name']) for s in suppliers]
```
Comment (idavidmcdonald) [inline app/main/forms/awards.py]: This form reads like we are passing a supplier ID and a supplier name. However in reality we are passing in a brief response ID and a supplier name for that brief response. We should make this clear

File: app/main/__init__.py (hunk 4 above)
Comment (katstevens) [inline app/main/__init__.py]: Now added

**Pull request merged**